src/rewrite_reduce.cpp

```diff
@@ -54,9 +55,59 @@ struct find_softmax
     }
 };
 
+struct find_reduce_mean
+{
+    auto matcher() const { return match::name("reduce_mean"); }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto reduce_mean = r.result;
+        auto op          = reduce_mean->get_operator().to_value();
+        auto axes        = op["axes"].to_vector<std::int64_t>();
+        auto input       = reduce_mean->inputs().front();
+
+        bool is_integral = false;
+        double max_n     = 0;
+        input->get_shape().visit_type([&](auto t) {
+            is_integral = t.is_integral();
+            max_n       = t.max();
+        });
+
+        auto dims = input->get_shape().lens();
+        double n  = std::accumulate(
```

Comment:
This can be calculated as `auto n = input->get_shape().elements() / reduce_mean->get_shape().elements()`.